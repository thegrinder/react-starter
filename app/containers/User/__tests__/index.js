// import React from 'react';

// import { render } from '../../../helpers/test-utils';
// import { User } from '..';

// const id = 'id';
// const fetchUser = jest.fn();
// const requiredProps = {
//   fetchUser,
//   loading: false,
//   match: {
//     params: {
//       id,
//     },
//   },
//   user: {
//     name: 'name',
//     username: 'username',
//     email: 'email@email.com',
//     phone: '132456776',
//   },
// };

// const renderComponent = (props = {}) => render(
//   <User {...requiredProps} {...props} />,
// );

// describe('<User />', () => {
//   it('should render correctly', () => {
//     const { container: { firstChild } } = renderComponent();
//     expect(firstChild).toBeDefined();
//     expect(firstChild).toMatchSnapshot();
//   });

//   it('should render the spinner if loading flag set to true', () => {
//     const { container: { firstChild }, getByLabelText } = renderComponent({ loading: true });
//     const spinnerElement = getByLabelText('loading...');
//     expect(firstChild).toContainElement(spinnerElement);
//   });

//   it('should render the spinner if user is not defined', () => {
//     const { container: { firstChild }, getByLabelText } = renderComponent({ user: undefined });
//     const spinnerElement = getByLabelText('loading...');
//     expect(firstChild).toContainElement(spinnerElement);
//   });

//   it('should fetch user with a specific id on mount', () => {
//     renderComponent();
//     expect(fetchUser).toHaveBeenCalledWith(id);
//   });
// });
