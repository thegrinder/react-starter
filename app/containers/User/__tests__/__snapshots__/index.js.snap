// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> should render correctly 1`] = `
<div
  class="sc-AxjAm PKtaI pv-6"
>
  <div
    class="sc-AxiKw kEQWPf pa-6"
  >
    <h1
      class="StyledHeading-fbinjh-0 fAjbYA"
      color="default"
    >
      name
    </h1>
    <ul
      class="max-w-sm"
    >
      <li
        class="flex justify-between"
      >
        <span
          class="StyledText-r7l7li-0 bMenBX"
          color="dark"
        >
          username:
        </span>
        <span
          class="StyledText-r7l7li-0 bklvHI"
          color="default"
        >
          username
        </span>
      </li>
      <li
        class="flex justify-between"
      >
        <span
          class="StyledText-r7l7li-0 bMenBX"
          color="dark"
        >
          email:
        </span>
        <span
          class="StyledText-r7l7li-0 bklvHI"
          color="default"
        >
          email@email.com
        </span>
      </li>
      <li
        class="flex justify-between"
      >
        <span
          class="StyledText-r7l7li-0 bMenBX"
          color="dark"
        >
          phone:
        </span>
        <span
          class="StyledText-r7l7li-0 bklvHI"
          color="default"
        >
          132456776
        </span>
      </li>
    </ul>
  </div>
</div>
`;
